src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/SnowplowEvent.scala

```diff
@@ -70,6 +80,18 @@ object SnowplowEvent {
       ).asJson
     }
 
+  implicit val unstructEventDecoder: Decoder[UnstructEvent] = deriveDecoder[UnstructEvent]
+
+  implicit val instantDecoder: Decoder[Instant] = Decoder.instance { c =>
+    c.as[String].flatMap { str =>
+      Either
+        .catchOnly[DateTimeParseException](Instant.parse(str))
+        .leftMap(e => DecodingFailure(e.getMessage, c.history))
+    }
+  }
+
+  implicit val eventDecoder: Decoder[Event] = deriveDecoder[Event]
```

Comment:
We need guarantees it will be able to decode an event encoded by default `Encoder`.

Reply:
Should I write tests for it or is there something else in your mind ?

Reply:
Tests should be enough. Don't forget about cases with:

* with empty contexts
* with non-empty contexts
* with unstructured event
* without unstructured event